samples/sdl_ttf/main.c

```diff
@@ -0,0 +1,241 @@
+
+/* Call this instead of exit(), so we can clean up SDL: atexit() is evil. */
+static void
+quit(int rc)
+{
+    SDL_VideoQuit();
+    _exit(rc);
+}
+
+SDL_Texture *
+LoadSprite(SDL_Renderer *renderer)
+{
+    SDL_Surface *temp;
+    SDL_Texture *sprite;
+    TTF_Font    *font;
+
+    SDL_Color    color = {0x7F, 0x7F, 0xFF, 0xFF};
+
+    font = TTF_OpenFont("pennaregular.ttf", 32);
+    if (font == NULL) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't load font: %s",
+                     SDL_GetError());
+        return NULL;
+    }
+
+    temp = TTF_RenderText_Blended(font, "NXDK", color);
+    if (temp == NULL) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't create surface: %s", SDL_GetError());
+        return NULL;
+    }
+
+    TTF_CloseFont(font);
+    /* Create textures from the image */
+    sprite = SDL_CreateTextureFromSurface(renderer, temp);
+    if (sprite == NULL) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't create texture: %s\n", SDL_GetError());
+        SDL_FreeSurface(temp);
+        return NULL;
+    }
+    SDL_FreeSurface(temp);
+
+    /* We're ready to roll. :) */
+    return sprite;
+}
+
+void
+MoveSprites(SDL_Renderer * renderer, SDL_Texture * sprite)
+{
+    int sprite_w, sprite_h;
+    int i;
+    SDL_Rect viewport;
+    SDL_Rect *position, *velocity;
+
+    /* Query the sizes */
+    SDL_RenderGetViewport(renderer, &viewport);
+    SDL_QueryTexture(sprite, NULL, NULL, &sprite_w, &sprite_h);
+
+    /* Draw a gray background */
+    SDL_SetRenderDrawColor(renderer, 0x00, 0x00, 0x00, 0xFF);
+    SDL_RenderClear(renderer);
+
+    /* Move the sprite, bounce at the wall, and draw */
+    for (i = 0; i < NUM_SPRITES; ++i) {
+        position = &positions[i];
+        velocity = &velocities[i];
+        position->x += velocity->x;
+        if ((position->x < 0) || (position->x >= (viewport.w - sprite_w))) {
+            velocity->x = -velocity->x;
+            position->x += velocity->x;
+        }
+        position->y += velocity->y;
+        if ((position->y < 0) || (position->y >= (viewport.h - sprite_h))) {
+            velocity->y = -velocity->y;
+            position->y += velocity->y;
+        }
+
+        /* Blit the sprite onto the screen */
+        SDL_RenderCopy(renderer, sprite, NULL, position);
+    }
+
+    /* Update the screen! */
+    SDL_RenderPresent(renderer);
+}
+
+void demo(void)
+{
+    int i, done;
+    const char *driver;
+    SDL_Window *window;
+    SDL_Renderer *renderer;
+    SDL_Texture *sprite;
+    int window_w, window_h;
+    int sprite_w, sprite_h;
+    SDL_Event event;
+
+    /* Enable standard application logging */
+    SDL_LogSetPriority(SDL_LOG_CATEGORY_APPLICATION, SDL_LOG_PRIORITY_INFO);
+
+    if (SDL_VideoInit(NULL) < 0) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't initialize SDL video: %s\n",
+                SDL_GetError());
+        _exit(1);
+    }
+
+    if (TTF_Init() < 0) {
```

Comment:
There should be a comment here or it should be part of a block which deals with font-texture creation, so people don't forget to copy it into their projects.